cinema/tests/test_movie_view_set.py

```diff
@@ -0,0 +1,125 @@
+from django.contrib.auth import get_user_model
+from django.test import TestCase
+from django.urls import reverse
+from rest_framework import status
+from rest_framework.test import APIClient, APITestCase
+
+from cinema.models import Movie, CinemaHall, Genre, Actor
+from cinema.serializers import (
+    MovieListSerializer,
+    MovieDetailSerializer,
+    MovieImageSerializer,
+    MovieSerializer,
+)
+from cinema.views import MovieViewSet
+
+MOVIE_URL = reverse("cinema:movie-list")
+MOVIE_SESSION_URL = reverse("cinema:moviesession-list")
+
+
+def sample_movie(**params):
+    defaults = {
+        "title": "Sample movie",
+        "description": "Sample description",
+        "duration": 90,
+    }
+    defaults.update(params)
+
+    return Movie.objects.create(**defaults)
+
+
+def sample_genre(**params):
+    defaults = {
+        "name": "Drama",
+    }
+    defaults.update(params)
+
+    return Genre.objects.create(**defaults)
+
+
+def sample_actor(**params):
+    defaults = {"first_name": "George", "last_name": "Clooney"}
+    defaults.update(params)
+
+    return Actor.objects.create(**defaults)
+
+
+def sample_movie_session(**params):
+    cinema_hall = CinemaHall.objects.create(name="Blue", rows=20, seats_in_row=20)
+
+    defaults = {
+        "show_time": "2022-06-02 14:00:00",
+        "movie": None,
+        "cinema_hall": cinema_hall,
+    }
+    defaults.update(params)
+
+
+class MovieTests(TestCase):
+
+    def setUp(self):
+        self.client = APIClient()
+        self.user = get_user_model().objects.create_user(
+            email="test@test.test", password="testpassword"
+        )
+        self.client.force_authenticate(self.user)
+
+
+class MovieViewSetTests(APITestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.user = get_user_model().objects.create_user(
+            email="test@test.test", password="testpassword"
+        )
+        self.client.force_authenticate(self.user)
+        self.genre = sample_genre()
+        self.actor = sample_actor()
+        self.movie = sample_movie()
+        self.movie.genres.add(self.genre)
+        self.movie.actors.add(self.actor)
+        self.movie.save()
+
+    def test_list_movies(self):
+        url = (MOVIE_URL)
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_retrieve_movie(self):
+        url = reverse("cinema:movie-detail", args=[self.movie.id])
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_filter_movies_by_title(self):
+        url = (MOVIE_URL) + f"?title={self.movie.title}"
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data), 1)
+        self.assertEqual(response.data[0]["title"], self.movie.title)
```

Comment:
Code Quality: In your `test_filter_movies_by_title` method, you are checking the status code of the response and the title of the first movie in the response data. It would be a good idea to also check the number of movies returned in the response data. You can do this by comparing the length of the response data to 1. This will ensure that only one movie is returned when filtering by title.